src/services/device-management/iot-device.service.ts

```diff
@@ -467,15 +468,54 @@ export class IoTDeviceService {
     }
 
     async delete(device: IoTDevice): Promise<DeleteResult> {
-        if (device.type == IoTDeviceType.LoRaWAN) {
-            const lorawanDevice = device as LoRaWANDevice;
-            this.logger.debug(
-                `Deleting LoRaWANDevice ${lorawanDevice.id} / ${lorawanDevice.deviceEUI} in Chirpstack ...`
-            );
-            await this.chirpstackDeviceService.deleteDevice(lorawanDevice.deviceEUI);
+        let deleteResult: DeleteResult = {
+            raw: null,
+            affected: 0
         }
 
-        return this.iotDeviceRepository.delete(device.id);
+        // If operations on chirpstack fail, rollback any mutations on the database
```

Comment:
"Run these operations inside a Typeorm transaction to make sure if operations on chirpstack fail, rollback any changes on the database to ensure consistency between devices in the OS2iot database and the Chirpstack database"